src/reference/90-Developers-Guide/02-Coding-Guideline/00.md

```diff
@@ -90,83 +90,87 @@ Use [MiMa][mima].
+The typeclass pattern with pure traits might be ease maintaining binary compatibility more so than subclassing.
 
 #### Avoid case classes, use sbt-datatype
 
-case class involves various codegen, which makes it harder to maintain binary compatibility over time.
+`case class` involves code generation that makes it harder to maintain binary compatibility over time.
 
 #### Prefer method overloading over default parameter values
 
-The default parameter values are effectively codegen,
-which makes it difficult to maintain.
+Default parameter values are effectively code generation,
+which makes them difficult to maintain.
 
-### Other public API matter
+### Other public API matters
 
-Here are other guidelines about public API.
+Here are other guidelines about the sbt public API.
 
 #### Avoid Stringly-typed programming
 
-Define datatypes.
+Define data types.
 
 #### Avoid overuse of `def apply`
 
 `def apply` should be reserved for factory methods
 in a companion object that returns type `T`.
 
-#### Use specific datatype (`Vector`, `List`, or `Array`) rather than `Seq`.
+#### Use specific data types (`Vector`, `List`, or `Array`), rather than `Seq`
 
 `scala.Seq` is `scala.collection.Seq`, which is not immutable.
 Default to `Vector`. Use `List` if constant prepending is needed.
-Use `Array` if Java interop is needed.
-Note using mutable collection is perfectly fine within the implementation.
+Use `Array` if Java interoperability is needed.
+Note that using mutable collections is perfectly fine within the implementation.
 
-#### Avoid calling `toSeq` or anything side-effecty on `Set`
+#### Avoid calling `toSeq` or anything with side-effects on `Set`
 
-`Set` is fine if you stick with set operations, like `contains` and `subsetOf`.
+`Set` is fine if you stick to set operations, like `contains` and `subsetOf`.
 More often than not, `toSeq` is called explicitly or implicitly,
-or some side effecting method is called from `map`.
+or some side-effecting method is called from `map`.
 This introduces non-determinism to the code.
 
 #### Avoid calling `toSeq` on `Map`
 
-Same as above. That will introduce non-determinism.
+Same as above. This will introduce non-determinism.
 
-#### Avoid functions and tuples in the signature, if Java interop is needed
+#### Avoid functions and tuples in the signature, if Java interoperability is needed
 
 Instead of functions and tuples, turn them into a trait.
-This is if Java interop is a concern, like implementing
-incremental compiler.
+This applies where interoperability is a concern, like implementing
+incremental compilation.
 
-### Style matter
+### Style matters
 
 #### Use scalariform
 
-sbt-houserules comes with scalariform.
+sbt-houserules comes with scalariform for formatting source code consistently.
 
 #### Avoid procedure syntax
 
-Return `Unit`.
+Declare an explicit `Unit` return.
+
+#### Define typeclass instances in companion objects
```

Comment:
Okay, becomes a little more wordy, but I think then the heading should be like:

> Define instances of typeclasses in their companion objects, when possible

Then, include an additional example of when it's okay/necessary *not* to do this.